app/helpers/taxonomy_helper.rb

```diff
@@ -135,4 +135,40 @@ def taxonomy_selects(f, selected_ids, taxonomy, label, options = {}, options_htm
     options[:label]    ||= _(label)
     multiple_selects f, label.downcase, taxonomy.authorized("assign_#{label.downcase}", taxonomy), selected_ids, options, options_html
   end
+
+  def all_checkbox(f, resource, label)
+    if User.current.admin? || User.current.filters.joins(:permissions).where({:'permissions.name' => "view_#{resource}",
+                                                       :search => nil,
+                                                       :taxonomy_search => nil}).present?
+      checkbox_f(f, :ignore_types,
+                   {:label => _("All") + label,
```

Comment:
"All #{label}" would be no better, as it can't be accurately translated.  It needs stating explicitly for each resource type.